### ISSUE TITLE
[Feature Request] Allow inline comments

### ISSUE THREAD

**Issue by lonix1** [logrotate/logrotate#390]
This is allowed:
````ini
# do this because of reason foo
notifempty
# do this because of reason bar
create 640 root root
````

This is not allowed:
````ini
notifempty                     # do this because of reason foo
create 640 root root           # do this because of reason bar
````

The second form is just as common when documenting config files, and much neater to read and maintain.

*Please consider supporting inline comments?*

And... Thanks for this really great tool!

**Comment by kdudka** [comment]
Thanks for the proposal.  I have not checked whether we would be able to implement this while preserving sufficient compatibility with existing configuration files.  Just let me highlight that logrotate is not the only program that does not support inline comments.  See for example the syntax of Apache's configuration files: https://httpd.apache.org/docs/2.4/configuring.html#syntax

**Comment by cgzones** [comment]
A starting point would be
```diff
diff --git a/config.c b/config.c
index 65b9262..1e9f3e7 100644
--- a/config.c
+++ b/config.c
@@ -161,7 +161,7 @@ static char *isolateLine(char **strt, char **buf, size_t length) {
 
     start = *strt;
     endtag = start;
-    while (endtag < max && *endtag != '\n') {
+    while (endtag < max && *endtag != '\n' && *endtag != '#') {
         endtag++;}
     if (max < endtag)
         return NULL;
```

One would need to take special attention about configuration directives possible taking `#` as part of a valid value:
```
compressoptions -#
olddir /var/my#log#dir/
```
and also within script definitions.
So some sort of quoting support might be necessary.

**Comment by lonix1** [comment]
@kdudka Yeah programs like apache don't do inline comments (though nginx does :wink: :wink: :wink:).

I'm not complaining, I think logrotate is great!

Keep in mind that inline commenting is common because it's easier to read and neater, and I find it keeps git diffs cleaner.
